File: Application/lib/model/episode.dart
import 'package:flutter_app_tv/model/genre.dart';
import 'package:flutter_app_tv/model/source.dart';
import 'package:json_annotation/json_annotation.dart';

class Episode{
  int id ;
  String title ;
  String downloadas ;
  String playas ;
  String description ;
  String duration ;
  String image ;



  List<Source> sources;


  Episode({
    this.id,
    this.title,
    this.downloadas,
    this.playas,
    this.description,

    this.duration,
    this.image,

    this.sources
  });

  factory Episode.fromJson(Map<String, dynamic> parsedJson){



    List<Source> _sources =  [];

    for(Map i in parsedJson ['sources']){
      Source source = Source.fromJson(i);
      if(source.kind != "download")
        _sources.add(source);
    }
    return Episode(
        id: parsedJson['id'],
        title : parsedJson['title'],
        downloadas : parsedJson['downloadas'],
        playas : parsedJson['playas'],
        description :(parsedJson['description'] == null)? "": parsedJson['description'],
        duration  : parsedJson['duration'],
        image  : parsedJson['image'],
        sources: _sources,

    );
  }

}

File: Application/android/app/src/main/kotlin/com/moontv/application/model/Episode.java
package com.moontv.application.model;

import java.util.List;
import com.google.gson.annotations.SerializedName;

public class Episode {

	@SerializedName("duration")
	private Object duration;

	@SerializedName("downloadas")
	private String downloadas;

	@SerializedName("image")
	private String image;

	@SerializedName("sources")
	private List<Source> sources;

	@SerializedName("playas")
	private String playas;

	@SerializedName("description")
	private String description;

	@SerializedName("id")
	private int id;

	@SerializedName("title")
	private String title;

	public Object getDuration(){
		return duration;
	}

	public String getDownloadas(){
		return downloadas;
	}

	public String getImage(){
		return image;
	}

	public List<Source> getSources(){
		return sources;
	}

	public String getPlayas(){
		return playas;
	}

	public String getDescription(){
		return description;
	}

	public int getId(){
		return id;
	}

	public String getTitle(){
		return title;
	}


	@Override
	public String toString() {
		return "Episode{" +
				"duration=" + duration +
				", downloadas='" + downloadas + '\'' +
				", image='" + image + '\'' +
				", sources=" + sources +
				", playas='" + playas + '\'' +
				", description='" + description + '\'' +
				", id=" + id +
				", title='" + title + '\'' +
				'}';
	}
}